### PR TITLE
#872 Add dark theme

### PR DESCRIPTION
**Pull request by pieer**

### CODE REVIEW

**Comment by mmbotelho** [comment]
Note: This PR closes #872

**Review by mmbotelho — DISMISSED** [review]
@pieer this is looking great! I found zero mistakes! 

However, there are things I'm not 100% happy with in terms of design, mainly how bright and colorful everything is. I really think I need to tone things down a bit and make them more neutral. But well, that's not related to your PR. 

Great job! 🎉

File: frontend/model/state.js [hunk 1]
```diff
@@ -348,6 +348,7 @@ const getters = {
     return Colors[state.theme]
   },
   isDarkTheme (state) {
+    if (!state.theme) state.theme = 'light' // TODO: check color available, prevent setting something differrent than available colors
```
Comment (sandrina-p) [inline frontend/model/state.js]: Do you plan to do this in this PR? If not, open an issue to not be forgotten or update the current one.

File: frontend/views/components/AppStyles.vue [hunk 2]
```diff
@@ -43,6 +43,9 @@ export default {
   computed: {
     colors () {
       return this.$store.getters.colors
+    },
+    lighterOrdDarker () {
```
Comment (sandrina-p) [inline frontend/views/components/AppStyles.vue]: `softTone` is a good hybrid name that adapts regardless of the theme.

**Review by sandrina-p — DISMISSED** [review]
Nothing to point out regarding the code! Awesome work!

File: frontend/model/state.js [hunk 3]
```diff
@@ -53,6 +53,13 @@ function guardedSBP (sel: string, ...data: any): any {
   return sbp(sel, ...data)
 }
 
+function checkThemeState (state) {
+  // Prevent bug if for some reason state doesn't have a theme
+  // TODO replace default from 0 to system preference if it exist
+  // Will be implemetaed in issue #844 Implement updates to the User Settings page
+  if (!state.theme) state.theme = Colors[0].theme
+}
```
Comment (taoeffect) [inline frontend/model/state.js]: I don't understand... how is it possible for the state to not have a theme? I don't think this is necessary. Maybe it was a fluke?

File: frontend/views/pages/DesignSystem.vue [hunk 4]
```diff
@@ -9,6 +9,7 @@ page(
     | A design system exists to help you build more cohesive interfaces.
     br
     | Follow the guidelines for the most common elements: Typography, Spacing, Colors, etc...
+    button.is-small.dark-team-toggle(@click='toggleTheme') Toggle {{isDarkTheme ? 'Dark' : 'Light' }} Theme
```
Comment (taoeffect) [inline frontend/views/pages/DesignSystem.vue:12]: Need `L` usage here, using a computed property for a complete sentence.

File: frontend/views/pages/DesignSystem.vue [hunk 5]
```diff
@@ -1350,7 +1350,16 @@ export default {
       return new Promise((resolve, reject) => {
         setTimeout(resolve, 2500)
       })
+    },
+    toggleTheme () {
+      this.setTheme(this.isDarkTheme ? 'light' : 'dark')
```
Comment (taoeffect) [inline frontend/views/pages/DesignSystem.vue]: It's possible that the issue with `checkThemeState` above is due to a typo. Can you convert all these string literals to be string constants? E.g. define in some file `const THEME_LIGHT = 'light'` (same for dark) and import and use it here and everywhere else `'light'` and `'dark'` appear.

**Review by taoeffect — CHANGES_REQUESTED** [review]
See comments! I wasn't able to test out the theme also because when I clicked to change it to dark theme nothing happened:

Screen Shot 2020-05-06 at 1 53 29 PM

**Comment by sandrina-p** [comment]
@taoeffect that's strange... I can test it locally. Did you see any error or so on the console?

**Comment by taoeffect** [comment]
@sandrina-p no console errors, done in Firefox Developer Edition, using commit b780d43137dc501cdfdc53397861b15eb672936b, which does appear to be the latest shown here on GH.

**Comment by sandrina-p** [comment]
@taoeffect, you are right. I have locally the first version of this PR. @pieer added two commits after my review and it broke the theme toggle logic. I tried to fix it so you can review it. Here's the fix:

```js
// state.js L192
setTheme (state, color) {
    state.theme = color
},
```

After that, it should work.

**Comment by taoeffect** [comment]
Good find! Yeah I think the `checkThemeState` stuff can be removed too.

**Review by sandrina-p — APPROVED** [review]

**Review by sandrina-p — APPROVED** [review]

**Review by taoeffect — APPROVED** [review]
Fantastic work @pieer! 🎉

**Pull request merged**